src/components/dialogs/create-study-dialog/directory-select.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { Button, Typography, Grid } from '@mui/material';
+import { FormattedMessage, useIntl } from 'react-intl';
+import DirectorySelector from '../directory-selector';
+import { setActiveDirectory } from '../../../redux/actions';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchPath } from '../../../utils/rest-api';
+
+interface DirectorySelectProps {
+    types: string[];
+}
+
+const DirectorySelect: React.FunctionComponent<DirectorySelectProps> = ({
+    types,
+}) => {
+    const intl = useIntl();
+    const dispatch = useDispatch();
```

Comment:
Why do you keep this in redux? The point is to move _all the form_ in react-hook-form.
Please migrate this component to RHF too.